### PR TITLE
Update to Swift 3

### PR DESCRIPTION
**Pull request by eddiekaiger**

### CODE REVIEW

File: TableSchemer/AccordionScheme.swift [hunk 1]
```diff
@@ -8,40 +8,40 @@
 
 import UIKit
 
-public class AccordionScheme<CollapsedCellType: UITableViewCell, ExpandedCellType: UITableViewCell>: BasicScheme<CollapsedCellType> {
+open class AccordionScheme<CollapsedCellType: UITableViewCell, ExpandedCellType: UITableViewCell>: BasicScheme<CollapsedCellType> {
     
-    public typealias AccordionConfigurationHandler = (cell: ExpandedCellType, index: Int) -> Void
-    public typealias AccordionSelectionHandler = (cell: ExpandedCellType, scheme: AccordionScheme, selectedIndex: Int) -> Void
+    public typealias AccordionConfigurationHandler = (_ cell: ExpandedCellType, _ index: Int) -> Void
```
Comment (ketzusaka) [inline TableSchemer/AccordionScheme.swift]: Is the `_` here part of adopting S3 api design guidelines? Previously we had the user put in `cell` and `index`.

File: TableSchemer/Scheme.swift [hunk 2]
```diff
@@ -55,7 +56,7 @@ public protocol Scheme: class {
         - parameter   cell:          The UITableViewCell being created.
         - parameter   relativeIndex: The cell index from the start of the scheme being configured.
      */
-    func configureCell(cell: UITableViewCell, withRelativeIndex relativeIndex: Int)
+    func configureCell(_ cell: UITableViewCell, withRelativeIndex relativeIndex: Int)
```
Comment (ketzusaka) [inline TableSchemer/Scheme.swift]: Do you think `configure(cell: UITableViewCell, withRelativeIndex relativeIndex: Int)` is more swift-3y?

File: TableSchemer/Scheme.swift [hunk 3]
```diff
@@ -71,7 +72,7 @@ public protocol Scheme: class {
         - parameter                 rowsBeforeScheme:   The number of rows before the scheme's first cell.
         - parameter                 relativeIndex:      The index of the row from the scheme's first cell.
      */
-    func selectCell(cell: UITableViewCell, inTableView tableView: UITableView, inSection section: Int, havingRowsBeforeScheme rowsBeforeScheme: Int, withRelativeIndex relativeIndex: Int)
+    func selectCell(_ cell: UITableViewCell, inTableView tableView: UITableView, inSection section: Int, havingRowsBeforeScheme rowsBeforeScheme: Int, withRelativeIndex relativeIndex: Int)
```
Comment (ketzusaka) [inline TableSchemer/Scheme.swift]: And this too, `select(cell: `

File: TableSchemer/SchemeSet.swift [hunk 4]
```diff
@@ -14,13 +14,13 @@
  *    have more than a single row), and the name property is used for the
  *    section name.
  */
-public class SchemeSet {
+open class SchemeSet {
```
Comment (ketzusaka) [inline TableSchemer/SchemeSet.swift]: Thinking more about this one, I feel its best `public`

File: TableSchemer/SchemeSetBuilder.swift [hunk 5]
```diff
@@ -30,32 +30,33 @@ public final class SchemeSetBuilder {
...
-        to the `TableScheme` automatically.
-
-        - parameter     handler:    The closure to configure the scheme.
-        - returns:                  The created Scheme instance.
+    - parameter     handler:    The closure to configure the scheme.
+    - returns:                  The created Scheme instance.
      */
-    public func buildScheme<BuilderType: SchemeBuilder>(@noescape handler: (builder: BuilderType, inout hidden: Bool) -> Void) -> BuilderType.SchemeType {
+    @discardableResult
+    public func buildScheme<BuilderType: SchemeBuilder>(_ handler: (_ builder: BuilderType, _ hidden: inout Bool) -> Void) -> BuilderType.SchemeType {
```
Comment (ketzusaka) [inline TableSchemer/SchemeSetBuilder.swift]: I think having the parameter names is more swifty 3y?

File: TableSchemer/TableScheme.swift [hunk 6]
```diff
@@ -263,7 +263,7 @@ public class TableScheme: NSObject {
         - parameter     tableView:       The UITableView to perform the animations on.
         - parameter     rowAnimation:    The type of animation that should be performed.
     */
-    public func reloadScheme(scheme: Scheme, inTableView tableView: UITableView, withRowAnimation rowAnimation: UITableViewRowAnimation = .Automatic) {
+    public func reloadScheme(_ scheme: Scheme, in tableView: UITableView, with rowAnimation: UITableViewRowAnimation = .automatic) {
```
Comment (ketzusaka) [inline TableSchemer/TableScheme.swift]: would `reload(scheme:` be more swift 3ish?

File: TableSchemer/Scheme.swift (hunk 2 above)
Comment (eddiekaiger) [inline TableSchemer/Scheme.swift]: I believe the verb and noun should go on the outside for Swift 3. Looking at the revised `UITableView` class, the methods adopt a similar style:

`open func insertSections(_ sections: IndexSet, with animation: UITableViewRowAnimation)`

File: TableSchemer/AccordionScheme.swift (hunk 1 above)
Comment (eddiekaiger) [inline TableSchemer/AccordionScheme.swift]: You actually get the following error if you try to put an argument label there:

`Function types cannot have argument label '[someLabel]'; use '_' insetad`

File: TableSchemer/SchemeSetBuilder.swift (hunk 5 above)
Comment (eddiekaiger) [inline TableSchemer/SchemeSetBuilder.swift]: Same error as in previous comment: 

`Function types cannot have argument label '[someLabel]'; use '_' instead`

**Pull request merged**